docs/book/src/bigint/index.md

```diff
@@ -0,0 +1,142 @@
+# Big Integers Library
+
+<!--Include BigInt when BigInt is available-->
+The Big Integers library provides a library to use extremely large numbers in Sway. It has 1 distinct types: `BigUint`. These types are heap allocated.
+
+Internally the library uses the `Vec<u64>` type to represent the underlying values of the big integers.
+
+For implementation details on the Big Integers Library please see the [Sway Libs Docs](https://fuellabs.github.io/sway-libs/master/sway_libs/bigint/index.html).
+
+## Importing the Big Integers Library
+
+In order to use the Big Integer Number Library, Sway Libs must be added to the `Forc.toml` file and then imported into your Sway project. To add Sway Libs as a dependency to the `Forc.toml` file in your project please see the [Getting Started](../getting_started/index.md).
+
+To import the Big Integer Number Library to your Sway Smart Contract, add the following to your Sway file:
+
+```sway
+{{#include ../../../../examples/big_integers/src/main.sw:import}}
+```
+
+In order to use any of the Big Integer types, import them into your Sway project like so:
+
+```sway
+{{#include ../../../../examples/big_integers/src/main.sw:import_big_uint}}
+```
+
+## Basic Functionality
+
+<!--Uncomment when BigInt is available-->
+<!--All the functionality is demonstrated with the `BigUint` type, but all of the same functionality is available for the other types as well.-->
+
+### Minimum and Maximum Values
+
+For the `BigUint` type, the minimum value is zero. There is no maximum value and the `BigUint` will grow to accommodate as large of a number as needed.
+
+### Instantiating a Big Integer
+
+#### Zero value
+
+Once imported, a `Big Integer` type can be instantiated defining a new variable and calling the `new` function.
+
+```sway
+{{#include ../../../../examples/big_integers/src/main.sw:initialize}}
+```
+
+this newly initialized variable represents the value of `0`.
+
+The `new` function is functionally equivalent to the `zero` function.
+
+```sway
+{{#include ../../../../examples/big_integers/src/main.sw:zero}}
+```
+
+<!--#### Positive and Negative Values
```

Comment:
Can this be removed or is the intention to add this in this PR?

Reply:
This is left in preparation of the `BigInt` type